libs/mpi/src/mpi_future.cpp

```diff
@@ -97,13 +97,9 @@ namespace hpx { namespace mpi { namespace experimental {
             // place this future data request in our queue for handling
             get_request_queue().enqueue(data);
 
-            // for debugging only
-            if (mpi_debug.is_enabled())
-            {
-                mpi_debug.debug(debug::str<>("request queued"), get_mpi_info(),
-                    "request", debug::hex<8>(data->request_), "active futures",
-                    debug::dec<3>(get_active_futures().size()));
-            }
+            mpi_debug.debug(debug::str<>("request queued"), get_mpi_info(),
+                "request", debug::hex<8>(data->request_), "active futures",
+                debug::dec<3>(get_active_futures().size()));
```

Comment:
I initially added the conditional as otherwise the arguments to the `.debug()` function would be evaluated even if the output was disabled. I assume you rely on the compiler to optimize away everything. Is that a 100% safe assumption?

Reply:
I already answered this comment, but github seems to have lost it - the conditional should not be necessary because the debug statement is a no-op when disabled. My experiments with godbolt indicate that all the code is removed, however I will conduct more tests just to reassure myself that no side effects are possible. A small unit test that flexes the debug::print code might prove useful to ensure that functions used in disabled call parameters are never invoked...